src/Microsoft.AspNet.Mvc.TagHelpers/LabelTagHelper.cs

```diff
@@ -45,12 +44,22 @@ public override void Process(TagHelperContext context, TagHelperOutput output)
                 {
                     output.MergeAttributes(tagBuilder);
 
+                    var childContent = await context.GetChildContentAsync();
+
                     // We check for whitespace to detect scenarios such as:
                     // <label for="Name">
                     // </label>
-                    if (string.IsNullOrWhiteSpace(output.Content))
+                    if (string.IsNullOrEmpty(output.Content))
```

Comment:
let's rope in @DamianEdwards tomorrow.  these tag helpers generally do not override what previous ones in the pipeline have done.  diverging from that here.